workers/unity/Assets/Fps/Scripts/Editor/GameLogic/WorldTiles/MapQualitySettingsInspector.cs

```diff
@@ -0,0 +1,204 @@
+
+            EditorGUI.BeginChangeCheck();
+
+            RefreshQualityProperties();
+            DrawPropertiesExcluding(serializedObject,
+                serializedObject.FindProperty("Settings").name);
+
+            GUILayout.Space(10);
+
+            GUILayout.Label("Tile draw distances", EditorStyles.boldLabel);
+            DrawTileQualityProperties();
+
+            GUILayout.Space(10);
+
+            if (!Application.isPlaying
+                || Application.isPlaying && MapQualitySettings.Instance == settings)
+            {
+                if (DrawPreviewToggle())
+                {
+                    DrawQualityPreviewButtons();
+                }
+            }
+
+            if (EditorGUI.EndChangeCheck())
+            {
+                EditorUtility.SetDirty(target);
+                settings.Apply();
+            }
+        }
+
+        private void RefreshQualityProperties()
+        {
+            if (QualitiesPropertyMatchesProject())
+            {
+                return;
+            }
+
+            var newQualities = new List<MapQualityLevelData>();
+            foreach (var qualityName in QualitySettings.names)
+            {
+                var newTileQualitySetting = new MapQualityLevelData();
+                newTileQualitySetting.QualityName = qualityName;
+                newTileQualitySetting.CheckoutDistance = MapQualitySettings.DefaultCheckoutDistance;
+                foreach (var setting in settings.Settings)
+                {
+                    if (setting.QualityName != qualityName)
+                    {
+                        continue;
+                    }
+
+                    newTileQualitySetting.CheckoutDistance = setting.CheckoutDistance;
+                    break;
+                }
+
+                newQualities.Add(newTileQualitySetting);
+            }
+
+            settings.Settings = newQualities;
+        }
+
+        private bool QualitiesPropertyMatchesProject()
+        {
+            if (settings.Settings.Count != QualitySettings.names.Length)
+            {
+                return false;
+            }
+
+            for (var i = 0; i < QualitySettings.names.Length; i++)
+            {
+                var name1 = settings.Settings[i].QualityName;
+                var name2 = QualitySettings.names[i];
+                if (name1 != name2)
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        private void DrawTileQualityProperties()
+        {
+            for (var i = 0; i < settings.Settings.Count; i++)
+            {
+                var qualityName = settings.Settings[i].QualityName;
+                var checkoutDistance = settings.Settings[i].CheckoutDistance;
+
+                GUI.color = (
+                        Application.isPlaying && MapQualitySettings.Instance == settings
+                        || MapQualitySettings.ShowPreview)
+                    && activeQualityLevel != i
```

Comment:
This conditional is hard to grok at first glance

Reply:
Not sure how this could be refactored though 😕

Reply:
Could turn it into a nicely named function or property, maybe? like... 

`ShouldHighlightCurrentQualityValue => Application.isPlaying && MapQualitySettings.Instance == settings || MapQualitySettings.ShowPreview`

I guess it's just moving the ungrokkable conditional about, but at least it explains what it's doing then :)

Reply:
Is now a property, `SettingsAreCurrentlyInUse`